risc0/zkvm/src/receipt.rs

```diff
@@ -326,3 +311,54 @@ impl Receipt {
         bytemuck::cast_slice(self.seal.as_slice())
     }
 }
+
+#[derive(Debug)]
+struct SystemState {
+    _pc: u32,
+    image_id: Digest,
+}
+
+#[derive(Debug)]
+struct Global {
+    pre: SystemState,
+    _post: SystemState,
+    _check_dirty: u32,
+    output: Digest,
+}
+
+impl SystemState {
```

Comment:
This is supposed to match the layout in the circuit I'm guessing?  I think this should really come from the exported layout so we don't have to worry if stuff shifts around.  But I guess I can do that later.

Reply:
I'd also love to have serialization (not just deserialization). Generally I think we need to improve the way that the host interacts with the globals, but this is a first tiny step in that direction.